guide/src/reference/attributes/on-rust-exports/main.md

```diff
@@ -0,0 +1,37 @@
+# `main`
+
+When attached to the `fn main()` function this attribute will adjust it to
+properly throw errors if they can be. This is only intended to be used for
+binaries.
+
+```rust
+#[wasm_bindgen(main)]
+fn main() -> Result<(), JsValue> {
+    Err(JsValue::from("this error message will be thrown"))
+}
+```
+
+The attribute also allows using `async fn main()` in Cargo binaries.
+
+```rust
+#[wasm_bindgen(main)]
+async fn main() {
+    // ...
+    future.await;
+}
+```
+
+Unlike `#[wasm_bindgen(start)]` this will not export a function to be executed
+on startup, it should only be used in Cargo binaries or examples for the `main`
+function. `#[wasm_bindgen(start)]` will prevent the `main` function to start and
```

Comment:
I'm not quite sure what this line is trying to say? I believe that combining them will actually cause the CLI to throw an error.

Reply:
What I'm trying to say is that using `#[wasm_bindgen(main)]` doesn't actually call that function when using the `init()` function. It only works as expected on `fn main` in a binary or example.

I will try to improve this, English isn't my first language, please feel free to suggest better wording.